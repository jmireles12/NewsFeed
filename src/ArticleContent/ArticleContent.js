import React, { Component } from 'react'
import ApiContext from '../ApiContext'
import Articles from '../Articles/Articles'
import { FindArticle } from '../news-helper'
import './ArticleContent.css'

export default class ArticleContent extends Component {
    
    static defaultProps = {
        match: {
            params: {}
        }
    }

    static contextType = ApiContext;
    render() {
        const { news=[] } = this.context
        const { articleId } = this.props.match.params
        const article = FindArticle(news, articleId) || { body: '' }
        console.log(this.props.id)
        return (
            <section className='ArticleContent'>
                <Articles
                    id={article.id}
                    name={article.name}
                />
                <div className='ArticleContent__content'>
                    {article.body.split(/\n \r|\n/).map((para, i) =>
                        <p key={i}>{para}</p>
                    )}
                </div>
            </section>
        )
    }
}