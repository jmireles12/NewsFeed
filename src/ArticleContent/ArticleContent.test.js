import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ArticleContent from './ArticleContent'

describe(`ArticleContent component`, () => {
    it('renders the complete ArticleContent', () => {
        const wrapper = shallow(<ArticleContent />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})