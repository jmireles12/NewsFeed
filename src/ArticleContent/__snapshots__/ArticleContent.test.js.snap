// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleContent component renders the complete ArticleContent 1`] = `
<section
  className="ArticleContent"
>
  <Articles
    match={
      Object {
        "params": Object {},
      }
    }
  />
  <section
    className="ArticleContent__content"
  >
    <p
      key="0"
    />
  </section>
</section>
`;
